xarray/core/missing.py

```diff
@@ -82,9 +84,11 @@ def __init__(self, xi, yi, method="linear", fill_value=None, period=None):
         self._xi = xi
         self._yi = yi
 
+        nan = np.nan if not complex_data else np.nan + np.nan * 1j
```

Comment:
can we use `yi.data.dtype.kind == "c"` instead of a new kwarg? or `np.iscomplexobj(yi)` as long as it doesn't trigger a dask compute

Reply:
yes